pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md

```diff
@@ -0,0 +1,116 @@
+---
+type: tutorial
+layout: tutorial
+title:  "Mapping Primitive Data Types from C"
+description: "Primitive Data types from C and how they look in Kotlin/Native side"
+authors: Eugene Petrenko 
+date: 2018-07-11
+showAuthorInfo: false
+issue: EVAN-5343
+---
+
+In this tutorial we learn how C data types are visible in Kotlin/Native and vice versa. You will: 
+- See what [Data Types are in C Language](#types-in-c-language)
+- Create a [tiny C Library](#example-c-library) that uses those types in exports
+- [Inspect Generated Kotlin/Native APIs from a C library](#inspecting-generated-kotlinnative-apis-for-a-c-library)
+- Find how [Primitive Types in Kotlin/Native](#primitive-types-in-kotlinnative) are mapped to C
+
+## Types in C Language
+
+What types do we have in C language? Let's first all of them. I use the
+[C data types](https://en.wikipedia.org/wiki/C_data_types) article from Wikipedia.
+It shows the following types groups:
+- basic types: `char, int, float, double`
+- structures, unions, arrays
+- pointers
+- function pointers
+
+There are also more specific types, that we are not going to cover:
+- boolean type (from [C99](https://en.wikipedia.org/wiki/C99))
+- `size_t` and `ptrdiff_t` (also `ssize_t`)
+- fixed width integer types, e.g. `int32_t` or `uint64_t` (from [C99](https://en.wikipedia.org/wiki/C99))
+
+In C language one have several modifiers like `signed, unsigned, short, long`. And type qualifiers like 
+`const`, `volatile`, `restruct`, `atomic`. We will not be discussing it here too.
+
+The best way to see how C data types are visible in Kotlin/Native is to try it. We create a 
+C library using all those types and see how Kotlin/Native uses it. You may check 
+[Interop with C Libraries](interop-with-c.html) tutorial for more details on how 
+Kotlin/Native interop works. In this tutorial we focus more on how types are mapped.  
+
+
+## Example C Library
+
+We create a `lib.h` file to demonstrate function signatures:
+```c
+#ifndef LIB2_H_INCLUDED
+#define LIB2_H_INCLUDED
+
+void ints(char c, short d, int e, long f);
+void doubles(float a, double b);
+
+#endif
+```
+
+The file is missing the `extern "C"` block, which is not needed for our example, by may be 
+necessary if you use C++ and overloaded functions. For more details, you may check the 
+[C++ compatibility](https://stackoverflow.com/questions/1041866/what-is-the-effect-of-extern-c-in-c)
+thread.
+
+It is only necessary to have an `.h` file to run the `cinterop` tool. And we do not need to create a 
+`lib.c` file, unless we want to compile and run the example.
+
+We still need to create a `.def` file to the `cinterop`. For more details
+you may check [Interop with C Libraries](interop-with-c.html). It is enough for
+the tutorial to create the `lib.def` file with the following content:
+
+```c
+headers = lib.h
```

Comment:
For this experiment you could just put C definitions directly to .def file, after 
`---` separator, i.e.
---
void ints(char c, short d, int e, long f);
void doubles(float a, double b);

Reply:
Nice trick! I will use it in the second tutorial to avoid repeating the same things